include/circt/Analysis/OpCountAnalysis.h

```diff
@@ -0,0 +1,43 @@
+//===- OpCountAnalysis.h - operation count analyses -----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This header file defines prototypes for methods that perform analysis
+// involving the frequency of different kinds of operations found in a
+// builtin.module.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_ANALYSIS_OPCOUNT_ANALYSIS_H
+#define CIRCT_ANALYSIS_OPCOUNT_ANALYSIS_H
+
+#include "circt/Support/LLVM.h"
+#include "mlir/IR/Operation.h"
+#include "llvm/ADT/DenseMap.h"
+
+namespace mlir {
+class AnalysisManager;
+} // namespace mlir
+namespace circt {
+namespace analysis {
+
+struct OpCountAnalysis {
```

Comment:
Nit: I think LLVM/MLIR coding guidelines say to only use a `struct` if all members are public. This would be better as a `class`.